receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_scraper_linux.go

```diff
@@ -7,13 +7,28 @@ package memoryscraper // import "github.com/open-telemetry/opentelemetry-collect
 
 import (
 	"github.com/shirou/gopsutil/v4/mem"
+	"go.opentelemetry.io/collector/featuregate"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/memoryscraper/internal/metadata"
 )
 
+var useMemAvailable = featuregate.GlobalRegistry().MustRegister(
+	"receiver.hostmetricsreceiver.UseLinuxMemAvailable",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterFromVersion("v0.134.0"),
+	featuregate.WithRegisterDescription("When enabled, the used value for the system.memory.usage and system.memory.utilization metrics will be based on the Linux kernel’s MemAvailable statistic instead of MemFree, Buffers, and Cached."),
+	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/42221"),
+)
+
 func (s *memoryScraper) recordMemoryUsageMetric(now pcommon.Timestamp, memInfo *mem.VirtualMemoryStat) {
-	s.mb.RecordSystemMemoryUsageDataPoint(now, int64(memInfo.Used), metadata.AttributeStateUsed)
+	// TODO: rely on memInfo.Used value once https://github.com/shirou/gopsutil/pull/1882 is released
```

Comment:
For easier reference can we point out that we do the same calculation that the gopsutil does until the release? https://github.com/shirou/gopsutil/pull/1882/files#diff-5af8322731595fb792b48f3c38f31ddb24f596cf11a74a9c37b19734597baef6R321